cmd/openshift-install/gather.go

```diff
@@ -123,8 +123,8 @@ func runGatherBootstrapCmd(directory string) error {
 func logGatherBootstrap(bootstrap string, port int, masters []string, directory string) error {
 	logrus.Info("Pulling debug logs from the bootstrap machine")
 	client, err := ssh.NewClient("core", net.JoinHostPort(bootstrap, strconv.Itoa(port)), gatherBootstrapOpts.sshKeys)
-	if err != nil && len(gatherBootstrapOpts.sshKeys) == 0 {
-		return errors.Wrap(err, "failed to create SSH client, ensure the proper ssh key is in your keyring or specify with --key")
+	if err != nil && strings.Contains(err.Error(), "ssh: handshake failed: ssh: unable to authenticate") {
```

Comment:
This error message doesn’t match the one in the bugzilla. Is there a distinction between when this error occurs vs the timeout message in the bz?

Reply:
There is for sure a distinction, and it's one of the things I addressed in this PR. The error in the BZ indicates that it was a timeout issue: `connect: operation timed out`. However, we still displayed the error about keys in the keyring, which became a red herring. This PR changes that behaviour to only show that error when auth failed. Furthermore, it makes the error more clear.

Reply:
I was under the impression that these timeout errors could occur do to ssh key problems and it seems that is what the bz indicates. I haven’t seen the authentication error before. So I am trying to understand when we see one vs the other.

Reply:
`connect: operation timed out` indicates specifically a failure to connect to the host:port at the `Transport Layer`. This could be because the service is not running, a firewall is blocking the port, or any number of other causes. Either way, the client will never reach the `Application Layer` and so the key is not even used. Still, we were previously mentioning that the key needed to be supplied when this happened. With this PR, such error will look like:

```
ERROR Attempted to gather debug logs after installation failure: failed to create SSH client: dial tcp 10.0.0.7:22: connect: operation timed out
```
When the authentication fails because of a bad or missing private key, we will always see `ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain`. Thus, why this PR looks for this string before outputting the `ensure the private key is added to your authentication agent (ssh-agent) or specified with the --key parameter` text in the error. With this PR, such an error will continue to say:
```
ERROR Attempted to gather debug logs after installation failure: ensure the private key is added to your authentication agent (ssh-agent) or specified with the --key parameter: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain
```